### PR TITLE
SNTWREC-49: Bucket Access

### PR DESCRIPTION
**Pull request by grusc**
With these changes
- the single ingest and delete endpoints now take in a pubsub event and reach out for the referenced blob in the bucket.
- the bulk ingest endpoint also scans the bucket specified in the request and collects all appropriate files

Also 
- added tests on the ingest service
- modified the docker compose to work with a provided service account secret

This branch was branched off from add-text-field-to-RE and rebased on dev

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    environment:
+      - API_PREFIX=
+      - BASE_URL_SEARCH=http://search-service
+      - BASE_URL_INGEST=http://embedding-service
+      - BASE_URL_EMBEDDING=http://embedding-service
+      - OPENSEARCH_HOST=
+      - OPENSEARCH_PORT=
+      - OPENSEARCH_USER=
+      - OPENSEARCH_PASS=
+      - OPENSEARCH_INDEX=
```
Comment (jesslou) [inline docker-compose.yaml:131]: Die OSS variables werden im Embedding Service nicht gebraucht, kann man der Übersicht halber weglassen.

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
```
Comment (jesslou) [inline docker-compose.yaml:83]: Hier gleich wie beim Embedding Service.

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      dockerfile: ./docker/Dockerfile
+    container_name: search-service
+    ports:
+      - 1336:80
+    expose:
+      - "1336"
+    environment:
+      - API_PREFIX=
+      - BASE_URL_SEARCH=http://search-service
+      - BASE_URL_INGEST=http://embedding-service
```
Comment (jesslou) [inline docker-compose.yaml:101]: Typo: BASE_URL_INGEST=http://ingest-service

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      dockerfile: ./docker/Dockerfile
+    container_name: embedding-service
+    ports:
+      - 1338:80
+    expose:
+      - "1338"
+    environment:
+      - API_PREFIX=
+      - BASE_URL_SEARCH=http://search-service
+      - BASE_URL_INGEST=http://embedding-service
```
Comment (jesslou) [inline docker-compose.yaml:125]: same here

File: docker-compose.yaml (hunk 1 above)
Comment (grusc) [reply docker-compose.yaml]: done

**Review by jesslou — APPROVED** [review]

**Review by jesslou — APPROVED** [review]

**Pull request merged**